plotly/tools.py

```diff
@@ -1395,27 +1395,33 @@ def return_figure_from_figure_or_data(figure_or_data, validate_figure):
 
 class TraceFactory(dict):
 
-    # use as validate method for all traces (not just streamline)
     @staticmethod
-    def validate(x, y, u, v,
-                 scale=.1, density=1,
-                 arrow_scale=.3, **kwargs):
+    def validate_shared(x, y, u, v,
```

Comment:
:cow2: Just a suggestion. Would this be clearer as two functions?

```
def validate_positive_scalars(**kwargs):
    """
    Validates that all values given in key/val pairs are positive.

    Accepts kwargs to improve Exception messages.
    
    :raises: (PlotlyError) If any value is < 0 or raises.
    """
    for key, val in kwargs.items():
        try:
            if val <= 0:
                raise PlotlyError('{} must be > 0, got {}'.format(key, val)
        except TypeError:
            raise PlotlyError('{} must be a number, got {}'.format(key, val)

def validate_array_pair(**kwargs):
    """"
    Validates a pair of named arrays.

    Accepts kwargs to improve Exception messages.

    :raises: (PlotlyError) On failure.
    """"
    for key, val in kwargs.items():
        # do your checks here...
```

I think that after the refactorings, this `validate` method has gotten much simpler than the docstring suggests?

Again, just a :cow2: